docs/howtos/branch-builds.md

```diff
@@ -0,0 +1,35 @@
+# Branch builds
+
+Branch builds are a way to build and test Fenix using branches from `application-services`, `android-components`, and/or
+`fenix`.  iOS is not currently supported, although we may add it in the future.
```

Comment:
> iOS is not currently supported, although we may add it in the future.

In https://github.com/mozilla/application-services/pull/4960 we added a smoke test that supports pulling a non-main branch of iOS and running tests against it, it's not yet configured to run on CI (opened #4966 for it) but maybe worth mentioning?